internal/appsec/waf/waf.go

```diff
@@ -944,3 +968,41 @@ func cindexCharPtrArray(array **C.char, i int) *C.char {
 	base := uintptr(unsafe.Pointer(array))
 	return *(**C.char)(unsafe.Pointer(base + unsafe.Sizeof((*C.char)(nil))*uintptr(i)))
 }
+
+// Atomic reference counter helper initialized at 1 so that 0 is the special
+// value meaning that the object was released and is no longer usable.
+type atomicRefCounter uint32
```

Comment:
Let's keep in mind that we'll be able to use the new go1.19 sync/atomic `Uint32` once it becomes the minimum supported version :tada:

Reply:
That will happen in two years 😅